core/src/main/scala/com/salesforce/op/stages/impl/tuning/Splitter.scala

```diff
@@ -82,12 +79,23 @@ abstract class Splitter(val uid: String) extends SplitterParams {
    * @param data
    * @return Parameters set in examining data
    */
-  def preValidationPrepare(data: Dataset[Row]): Option[SplitterSummary]
-
+  def preValidationPrepare(data: Dataset[Row]): PrevalidationVal
 
   protected def checkPreconditions(): Unit =
     require(summary.nonEmpty, "Cannot call validationPrepare until preValidationPrepare has been called")
 
+  /**
+    * Add a splitter parameter to name the label column
+    * @param label
+    * @return
+    */
+  def withLabelColumnName(label: String): Splitter = {
+    if (!isSet(labelColumnName)) {
```

Comment:
so there is not way to override the label column once it's set? why limit it?

Reply:
I find it difficult to reason about the logic without making this param immutable once set.

Reply:
I think this will bite us in the future.

Reply:
No doubt, but I think it will bite us unpredictably if we make it mutable. I can add a warning log instead of silently refusing to modify it.